packages/hebao_v2/contracts/base/libwallet/GuardianLib.sol

```diff
@@ -42,8 +42,10 @@ library GuardianLib
         external
     {
         address guardian = address(0);
+        address owner = wallet.owner;
         for (uint i = 0; i < _guardians.length; i++) {
             require(_guardians[i] > guardian, "INVALID_ORDERING");
+            require(_guardians[i] != owner, "GUARDIAN_CAN_NOT_BE_OWNER");
```

Comment:
Add this check in `_addGuardian` because otherwise the owner could still be added using one of the other methods to add guardians?

Reply:
I guess best to remove the check here now because it's not needed anymore.

Reply:
Yes, It's hard to say which one is the better choice because the relayer should do the check. Do this check in the contract can reduce possibilities so prevent potential bugs.

Reply:
Oh yeah I can see the benefit. But I meant that this function also calls `_addGuardian` which will already do the check that the guardian isn't the owner now so doing this check in `addGuardiansImmediately` as well will just end up doing the same check twice. So it would make sense to remove the check here in `addGuardiansImmediately`.

Reply:
Oh, yes, I see.